windows/README_MINGW

```diff
@@ -0,0 +1,218 @@
+
+pacman -Suy
+
+- Install needed packages
+
+pacman -S mingw-w64-x86_64-toolchain patch autoconf make \
+gettext-devel automake libtool pkgconfig p7zip unzip git python
+
+- Replace the GCC compiler with a version configured for SJLJ exceptions, as
+instructed by the original Windows build instructions (windows/README)
+
+Prebuilt binaries can be found at
+https://sourceforge.net/projects/mingw-w64/files/Toolchains%20targetting%20Win64/
+
+Look for "x86_64-posix-sjlj" under "MinGW-W64 GCC-{version}", the file should be
+called x86_64-{version}-release-posix-sjlj-rt_v6-rev0.7z
+
+Or just download from a direct link (GCC 8.1.0):
+
+wget https://downloads.sourceforge.net/project/mingw-w64/Toolchains%20targetting%20Win64/Personal%20Builds/mingw-builds/8.1.0/threads-posix/sjlj/x86_64-8.1.0-release-posix-sjlj-rt_v6-rev0.7z
+
+Once downloaded:
+
+p7zip -d x86_64-8.1.0-release-posix-sjlj-rt_v6-rev0.7z
+
+That will decompress to a folder called mingw64 in the working directory.
+Now replace the files from the previous mingw-w64-x86_64-toolchain package
+installation:
+
+mv /mingw64 /mingw64.pacman
+mv mingw64 /
+
+Make sure GCC still runs and it is the version we need:
+
+$ gcc --version
+gcc.exe (x86_64-posix-sjlj-rev0, Built by MinGW-W64 project) {version}
+
+
+
+Preparing JACK dependencies
+---------------------------
+
+There are prebuilt MinGW binaries for all the libraries that can be installed
+using the pacman package manager, but since we are using a compiler that is not
+the default version shipped by MinGW, it seems better idea to build from source
+to avoid any linker and runtime issues. A good technical explanation and/or
+evidence for this statement is currently missing from this guide.
+
+Fortunately there are PKGBUILD files for doing so together with a nice guide at
+https://github.com/msys2/MINGW-packages
+
+git clone https://github.com/msys2/MINGW-packages.git
+
+cd MINGW-packages/mingw-w64-{libname}-git
+MINGW_INSTALLS=mingw64 makepkg-mingw -sLf
+pacman -U mingw-w64-{libname}-git-{suffix}.pkg.tar.xz
+
+Repeat procedure for each library replacing {libname} with the appropriate name
+and {suffix} with whatever the above process created.
+
+db
+libsndfile
+libsamplerate
+
+Some libraries like libsndfile and libsamplerate will ask for installing extra
+dependencies, it is ok to do so. The following need to be built in order:
+
+libtre
+libsystre
+
+libsystre is a wrapper around libtre that allows including <regex.h> later.
+Before building it adjust the includes path, it can be unset afterwards:
+
+export C_INCLUDE_PATH=/mingw64/include
+
+Finally for portaudio the Steinberg ASIO SDK should be manually downloaded.
+It can be found at https://www.steinberg.net/en/company/developers.html. The waf
+script will later check if the SDK is present at /opt/asiosdk
+
+wget https://www.steinberg.net/asiosdk -O /tmp/asiosdk.zip
+unzip /tmp/asiosdk.zip -d /tmp
+mv /tmp/asiosdk_{version}_{date} /opt/asiosdk
```

Comment:
I cannot recall if `/opt `is already present in a fresh MinGW install. `mkdir /opt/asiosdk -p `would help in if `/opt` is not already present but then the `mv` line will result in a possibly unnecessary level:

`/opt/asiosdk/asiosdk_{version}_{date}`

Also I am removing `{version}_{date}` on purpose so it is not carried to subsequent steps and also because the wscript checks at a hardcoded place` /opt/asiosdk`. Otherwise the user would have to adjust wscript depending the ASIO SDK minor version or date, or use a environment variable, etc.  I think this can be easily avoided by stripping the version/date suffixes right on this early filesystem step.

Reply:
`/opt` is not present in a fresh MinGW install. Because of the version path situation you explained, it might be sufficient to change the line to `mkdir /opt`.